Release 3.0.0-alpha.1 (#4861)
open-formulieren/open-forms: CHANGELOG.rst

    - Added the serializer option to allow selecting document type by description (omschrijving).
    - Deprecated ZGW url-based document type configuration.
* [:backend:`4431`] Improved addressNL mapping backwards compatibility.
* [:backend:`4418`] Made it possible to map specific addressNL fields or a whole object to ObjectsAPI.
* [:backend:`4693`] Integrated Objects API prefill modal with backend.
    - Added copy button to the modal, which allows the user to copy the configuration used by the Form's
      registration backend. This is shown only when backends are available.
    - Added dropdown to select a registration backend instead of always picking the first one.
* [:backend:`4745`] Added registration variable containing all attachment URLs.
* [:backend:`4796`] Added product url to ZGW APIs registration.
* [:backend:`4762`] Added cosign data (bsn) to the StUF-ZDS creerzaak bericht.
* [:backend:`4798`] Added custom confirmation modal, which replaces the old way of displaying confirmation
  messages.
* [:backend:`4344`] It's now possible to configure Objects API group in ZGW registration. This used to be by
  default the one with the lowest primary key.

> * [:backend:`4320`] Cosign improvements. 
>     - Updated translations so the suggested texts are already included.
>     - Added the ability to add a 'cosign now' button on confirmation page.
>     - Made the cosign verification flow faster when the code is included in the link in the email.
>     - Made the confirmation page title and body configurable.
>     - Updated cosign configuration docs and manual.

Reviewer: ```suggestion
* [:backend:`4320`] Improved the cosign flow and the texts used in cosign flows, while adding more flexibility:

    - You can now use templates specifically for cosigning for the confirmation screen content,
      with the ability to include a 'cosign now' button.
    - You can now use templates specifically for cosigning for the confirmation email subject and content.
    - When links are used in the cosign request email, the cosigner can now directly click through without
      having to enter a code to retrieve the submission.
    - Updated the default templates with better text/instructions.
```

* [:backend:`4815`] Changed submission removal limit to 0, allowing submissions to be deleted after 0 days
  (i.e. on the same day).

* [:backend:`4717`] Improved accessibility for site logo, error message element and PDF documents. 
* Replaced duplicated code for payment/registration plugin configuration option forms, by adding a generic

>   component.

Reviewer: Something more for "project maintenance", no?

> * Refactored demo plugins configuration to use a modal instead of inline configuration options.

Reviewer: ```suggestion
```
not very interesting, and the entry for #4686 can be generalized :)

> * [:backend:`4707`] Added the option to resize the json widget manually.    

Reviewer: ```suggestion
* [:backend:`4707`] You can now resize the Json-Logic widgets.
```

> * [:backend:`4686`] Improved UX for email and StUF-ZDS registrations by using React components, instead of
>   react-json-schema-form generated forms.

Reviewer: ```suggestion
* [:backend:`4686`} All the registration plugin configuration options are now consistently managed in a 
   modal with better UX.
```

* [:backend:`4720`] Improved accessibility for the skiplink and the PDF report.

> * [:backend:`4396`] Added Objects API prefill plugin.

Reviewer: maybe put this closer to the entry for #4693, and put both ticket numbers there?

**Bugfixes**

> * Fixed tests after removing password component.

Reviewer: ```suggestion
```
really belongs together with the change that the password component is removed tbh

> * [:backend:`4732`] Fixed CSP issues for Expoints/other analytics.

Reviewer: for the issues that were backported (and also added to the master branch changelog), it's best to just copy paste the exact same changelog entries from https://github.com/open-formulieren/open-forms/blob/master/CHANGELOG.rst?plain=1

* Fixed the logic rules manual regarding "duration" which did not work.
* [:backend:`4823`] Fixed uploading files with leading or trailing whitespace in the
  filename.

> * [:backend:`4810`] Fixed uppercase component variable values turing lowercase. See the

Reviewer: the remark is missing :) though I think it's better to update `src/openforms/upgrades/upgrade_paths.py` and require a minimum version of 2.8.2 before you can upgrade to 3.0, and then we can assume the fix script has been executed.

  remark above for additional instructions.
* Fixed recursion issues in component search strategies.
* [:backend:`4826`] Added a work around to fix the translation issue and to properly evaluate the
  translated default values for the newly added configuration model fields.

> * [:backend:`4772`] Fixed Select components' unset dataType which caused issues when submitting data to
>   the backend.

Reviewer: ```suggestion
* [:backend:`4772`] Fixed select components with integer values being treated as numbers
  instead of strings.
```

* [:backend:`4727`] Fixed crash when a user defined variable was changed to an array
  datatype.

> * [:backend:`4719`] Improved accessibility in postcode fields.

Reviewer: Bugfix? or new feature? :thinking: "Improved" reads as a new feature, otherwise it should be "Fixed accesibility..."

> * [:backend:`4802`] Fixed react-select styling issue regarding growing beyond its max inline size.

Reviewer: ```suggestion
* [:backend:`4802`] Fixed some dropdowns taking up more horizontal space than intended.
```

> * Fixed type error in object spread.

Reviewer: ```suggestion
* Fixed some pre-configured component configurations not being applied entirely when adding them to the form.
```

> * [:backend:`4763`] Fixed temporary file uploads that could not be deleted.

Reviewer: ```suggestion
* [:backend:`4763`] Fixed temporary file uploads not being delete-able in the admin interface.
```

* [:backend:`4726`] Fixed the styling for form delete buttons.

> * [:backend:`4546`] Ensured the soft required component integrates well.

Reviewer: ```suggestion
* [:backend:`4546`] Fixed the soft-required validation errors being shown in the summary PDF.
```

**Project maintenance**

> * [:backend:`4849`] Re-recorded VCR cassettes.

Reviewer: ```suggestion
```
I know it's a lot of work, but this isn't really useful in the changelog

> * Updated CHANGELOG with 2.7.10 and 2.8.2 patch releases changes.

Reviewer: ```suggestion
```
there's little added value for these kind of items

* Upgraded to MSW 2.x.
* Bumped formio-builder version.

> * [:backend:`4431`] Refactored the Objects API v2 handler according to the new mappings implementation.

Reviewer: ```suggestion
```
I consider it part of the AddressNL ticket, no need to mention this separately.

* [:backend:`3283`] Removed password Formio component.

> * Bumped lxml html cleaner.

Reviewer: you can summarize this one and the waitress one with a one liner like:

```suggestion
* Upgraded some dependencies to their latest security releases.
```
it's mostly signaling that we're on top of reported security issues in third party dependencies

* Dropped RJSF dependency.
* Bumped waitress 

> * Refactor shared utilities to real shared general purpose location.

Reviewer: ```suggestion
```
not much value in mentioning this
